src/main/kotlin/io/github/gradlenexus/publishplugin/NexusRepositoryContainer.kt

```diff
@@ -18,8 +18,22 @@ package io.github.gradlenexus.publishplugin
 
 import org.gradle.api.Action
 import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.NamedDomainObjectProvider
 
 interface NexusRepositoryContainer : NamedDomainObjectContainer<NexusRepository> {
-    fun sonatype(): NexusRepository
-    fun sonatype(action: Action<in NexusRepository>): NexusRepository
+    val sonatype: NamedDomainObjectProvider<out NexusRepository>
+        get() =
+            // See https://github.com/gradle/gradle/issues/8057#issuecomment-826933995, Introduce TaskContainer.maybeNamed
+            if ("sonatype" in names) {
+                named("sonatype")
+            } else {
+                register("sonatype")
+            }
+
+    @Deprecated("Use sonatype property instead", ReplaceWith("sonatype"))
+    fun sonatype(): NexusRepository = sonatype.get()
```

Comment:
Technically speaking, old syntax still works with Kotlin DSL.
However, Groovy DSL does not support `sonatype()` since, well, you know, the target for configure becomes `AbstractNamed...` thing, and it does not see this `sonatype()` function

Reply:
> Technically speaking, old syntax still works with Kotlin DSL.

Isn't that because Kotlin DSL never supported `container { name }` syntax? So, it only worked because DSL forces us to do `by creating` and other magic.

> Groovy DSL does not support sonatype() since, well, you know, the target for configure becomes AbstractNamed... thing, and it does not see this sonatype() function

Ah, of course, the root cause is that we need to remove that magical `configure` function.

Reply:
I also think keeping the old syntax with `()` as non-deprecated is a good idea.

However, I'm not sure, how would be best to emphasis that there are problematic to use in the Groovy DSL.